### PR TITLE
Add implementation to tablet

### PR DESCRIPTION
**Pull request by chaoyli**

### CODE REVIEW

**Review by kangpinghuang — CHANGES_REQUESTED** [review]

File: be/src/olap/tablet.cpp [hunk 1]
```diff
@@ -54,8 +56,62 @@ using std::vector;
...
+    std::string shard_str = shard_path.substr(shard_path.find_last_of('/') + 1);
+    uint64_t shard = stol(shard_str);
+    tablet_meta->set_shard_id(shard);
+
+    // save tablet_meta info to kv db
+    // tablet_meta key format: tablet_id + "_" + schema_hash
+    OLAPStatus res = TabletMetaManager::save(data_dir, tablet_meta->tablet_id(),
+                                             tablet_meta->schema_hash(), tablet_meta);
+    if (res != OLAP_SUCCESS) {
+        OLAP_LOG_WARNING("fail to save tablet_meta to db. [file_path=%s]", file_path.c_str());
```
[the remark this answers is not in the document]
Comment (chaoyli) [reply be/src/olap/tablet.cpp]: OK

File: be/src/olap/rowset/alpha_rowset_writer.cpp [hunk 2]
```diff
@@ -160,9 +161,9 @@ RowsetSharedPtr AlphaRowsetWriter::build() {
...
         }
     }
-    Rowset* rowset = new AlphaRowset(_rowset_writer_context.tablet_schema,
-                                     _rowset_writer_context.rowset_path_prefix,
-                                     _rowset_writer_context.data_dir, _current_rowset_meta);
+    RowsetSharedPtr rowset(new AlphaRowset(_rowset_writer_context.tablet_schema,
+                                    _rowset_writer_context.rowset_path_prefix,
+                                    _rowset_writer_context.data_dir, _current_rowset_meta));
     rowset->init();
     return std::shared_ptr<Rowset>(rowset);
```
[the remark this answers is not in the document]
Comment (chaoyli) [reply be/src/olap/rowset/alpha_rowset_writer.cpp]: OK

File: be/src/olap/tablet.cpp [hunk 3]
```diff
@@ -65,9 +121,54 @@ Tablet::~Tablet() {
...
 
+OLAPStatus Tablet::load() {
+    OLAPStatus res = OLAP_SUCCESS;
+    MutexLock l(&_load_lock);
+
+    if (_is_loaded) {
+        return OLAP_SUCCESS;
+    }
+
+    res = load_indices();
```
[the remark this answers is not in the document]
Comment (chaoyli) [reply be/src/olap/tablet.cpp]: OK

File: be/src/olap/tablet.cpp [hunk 4]
```diff
@@ -638,45 +744,213 @@ OLAPStatus Tablet::register_tablet_into_dir() {
...
+        LOG(FATAL) << "fail to save tablet_meta when delete expire incremental data."
+                   << "tablet=" << full_name();
+    }
+    remove_files(files_to_remove);
+}
+
+OLAPStatus Tablet::clone_data(const TabletMeta& tablet_meta,
+                              const std::vector<RowsetMetaSharedPtr>& rowsets_to_clone,
+                              const std::vector<Version>& versions_to_delete) {
+    LOG(INFO) << "begin to clone data to tablet. tablet=" << full_name()
```
[the remark this answers is not in the document]
Comment (chaoyli) [reply be/src/olap/tablet.cpp]: I will replace it using revise_tablet_meta.

File: be/src/olap/tablet.cpp [hunk 5]
```diff
@@ -503,6 +633,11 @@ void Tablet::list_versions(vector<Version>* versions) const {
     }
 }
 
+void Tablet::list_version_entities(vector<VersionEntity>* version_entities) const {
+    DCHECK(version_entities != nullptr && version_entities->empty());
+    return;
+}
+
```
[the remark this answers is not in the document]
Comment (chaoyli) [reply be/src/olap/tablet.cpp]: I will remove list_version_entities function.

**Pull request merged**